File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task167.py
from decimal import Decimal, ROUND_HALF_UP, getcontext
import math

class Task167:
    @staticmethod
    def cubeDiagonal(volume: float) -> float:
        if not math.isfinite(volume) or volume < 0.0:
            return float('nan')
        diag = (volume ** (1.0 / 3.0)) * math.sqrt(3.0)
        getcontext().prec = 28
        rounded = Decimal(str(diag)).quantize(Decimal('0.01'), rounding=ROUND_HALF_UP)
        return float(rounded)

if __name__ == "__main__":
    tests = [8.0, 343.0, 1157.625, 0.0, -5.0]
    for v in tests:
        res = Task167.cubeDiagonal(v)
        if math.isnan(res):
            print(f"cubeDiagonal({v}) = NaN")
        else:
            print(f"cubeDiagonal({v}) = {res:.2f}")

File: dataset/Vanilla/OPENAI_gpt-5-2025-08-07/Task167.c
#include <stdio.h>
#include <math.h>

double cubeDiagonal(double volume) {
    double d = cbrt(volume) * sqrt(3.0);
    return round(d * 100.0) / 100.0;
}

int main() {
    double tests[] = {8, 343, 1157.625, 1, 0.125};
    int n = sizeof(tests) / sizeof(tests[0]);
    for (int i = 0; i < n; i++) {
        double res = cubeDiagonal(tests[i]);
        printf("%.2f\n", res);
    }
    return 0;
}